### PR TITLE
DataGrid: Use the feature toggle to include this new panel

### PR DESCRIPTION
**Pull request by ryantxu**
Rather than have the feature toggle enable editing within a datagrid -- this changes it so it will show/hide the entire panel.  

It adds an explicit panel option for editing options.

### CODE REVIEW

File: public/app/plugins/panel/datagrid/DataGridPanel.tsx [hunk 1]
```diff
@@ -40,7 +40,13 @@ import {
 export interface DataGridProps extends PanelProps<PanelOptions> {}
 
 export function DataGridPanel({ options, data, id, fieldConfig, width, height }: DataGridProps) {
-  const [state, dispatch] = useReducer(datagridReducer, initialState);
+  const [state, dispatch] = useReducer(datagridReducer, {
+    ...initialState,
+
+    // TODO -- also check if the user has permissions?
+    // TODO -- respond to changes in the option
+    editingEnabled: options.enableEditing,
```
Comment (ryantxu) [inline public/app/plugins/panel/datagrid/DataGridPanel.tsx:48]: can useReducer respond to changers in the options?  (my redux skillz are zero)

File: pkg/api/frontendsettings.go [hunk 2]
```diff
@@ -62,6 +63,11 @@ func (hs *HTTPServer) getFrontendSettings(c *contextmodel.ReqContext) (*dtos.Fro
 			continue
 		}
 
+		// Hide the datagrid until we remove the feature flag
+		if panel.ID == "datagrid" && !hs.Features.IsEnabled(featuremgmt.FlagEnableDatagridEditing) {
```
Comment (mdvictor) [inline pkg/api/frontendsettings.go:67]: This works great if we want to hide the datagrid alltogether. But having both this and the `PanelDataErrorView` seems a bit overkill.

File: public/app/plugins/panel/datagrid/DataGridPanel.tsx (hunk 1 above)
Comment (mdvictor) [reply public/app/plugins/panel/datagrid/DataGridPanel.tsx:48]: yeah, that can be done. It's a cool new option to add, but I would leave it as a future improvement after the release

File: public/app/plugins/panel/datagrid/DataGridPanel.tsx [hunk 3]
```diff
@@ -208,6 +214,10 @@ export function DataGridPanel({ options, data, id, fieldConfig, width, height }:
     return <PanelDataErrorView panelId={id} fieldConfig={fieldConfig} data={data} />;
   }
 
+  if (!isDatagridEnabled()) {
+    return <PanelDataErrorView panelId={id} message="Datagrid is not enabled" fieldConfig={fieldConfig} data={data} />;
+  }
+
```
Comment (mdvictor) [inline public/app/plugins/panel/datagrid/DataGridPanel.tsx:220]: I feel like this is an elegant way of having the panel behind a ff. I don't know how it didn't hit me earlier :)) Thank you!

**Review by mdvictor — COMMENTED** [review]
Thank you so much for opening this PR. I think that the `PanelDataErrorView` is really nice because it shows a suggestive message to the user, while also leaving it in the plugin list so that people can see it and wonder what it's about.

I am thinking of pushing only that bit to main and leaving the `enableEditing` as an improvement for after G10

**Pull request closed without merging**